### PR TITLE
RHCLOUD-1295 - Add Entitlement check for hybrid bundle

### PR DESCRIPTION
**Pull request by tahmidefaz**
Basically working. Need to add caching ability.

### CODE REVIEW

**Comment by tahmidefaz** [comment]
@iphands I am not sure the way I solved this is acceptable or not. I had to edit existing unit tests to make this work. If I call checkHybrid() in subscriptions.go from outside getSubscriptions(), it fails all unit tests.

**Comment by tahmidefaz** [comment]
More work needs to be done after talking with Ian.

**Comment by tahmidefaz** [comment]
Please note that getSubscriptions() not returns a filtered array of skus of the user instead of the whole response body. I think this allows for a simpler sku checks for the hybrid and could be beneficial in the future as well.

File: controllers/subscriptions.go [hunk 1]
```diff
@@ -118,8 +150,11 @@ func Index(getCall func(string) (types.SubscriptionsResponse)) func(http.Respons
...
 
+		hybridSKUs := []string{"SVC3851", "SVC3852", "SVCSER0566", "SVCSER0567"}
+		entitleHybrid := len(checkCommon(hybridSKUs, res.Data)) > 0
+
 		obj, err := json.Marshal(types.EntitlementsResponse{
-			HybridCloud:    types.EntitlementsSection{IsEntitled: true},
+			HybridCloud:    types.EntitlementsSection{IsEntitled: entitleHybrid},
 			Insights:       types.EntitlementsSection{IsEntitled: entitleInsights},
 			Openshift:      types.EntitlementsSection{IsEntitled: true},
 			SmartMangement: types.EntitlementsSection{IsEntitled: (len(res.Data) > 0)},
```
Comment (ryelo) [inline controllers/subscriptions.go]: I think this line needs to be updated. Before, SKU3124 was the only SKU we were checking for, so we just wanted to see if it was there by checking the length. Now that we have more than one, I think it's possible that the length could be `>0` but not have SKU3124

Comment (tahmidefaz) [reply controllers/subscriptions.go]: Thank you so much for pointing that out! I wasn't sure what SKU3124 did. This should be fixed now!

**Pull request merged**